adding Elixir implementation

cdesch opened the pull request:
## Description

Adding an implementation in Elixir. This implementation uses recursion in place of a while loop since Elixir does not have a true while loop. Elixir is also immutable requiring the bit array in Dave's implementation to be transformed and returned.


## Contributing requirements

* [x ] I read the contribution guidelines in CONTRIBUTING.md.
* [x ] I placed my solution in the correct solution folder.
* [x ] I added a README.md with the right badge(s).
* [x ] I added a Dockerfile that builds and runs my solution.
* [x ] I selected `drag-race` as the target branch.
* [x ] All code herein is licensed compatibly with BSD-3.

rbergen commented on PrimeElixir/solution_1/README.md:
```diff
@@ -0,0 +1,34 @@
+# Ruby solution by rbergen
```
I feel honoured you used my Ruby README.md as the template for yours. I do think the title of this one should probably be updated, though :)

cdesch replied:
Thanks! I updated it!

rbergen commented on PrimeElixir/solution_1/README.md:
```diff
@@ -0,0 +1,34 @@
+# Ruby solution by rbergen
+
+![Algorithm](https://img.shields.io/badge/Algorithm-base-green)
+![Faithfulness](https://img.shields.io/badge/Faithful-no-green)
```
For a solution that is not faithful, the badge colour to use is -yellowgreen instead of -green

cdesch replied:
Thank you! Changed!

rbergen commented on PrimeElixir/solution_1/prime.ex:
```diff
@@ -0,0 +1,78 @@
+
+size = 1000000
+primes = Enum.map(1..size, fn(_x) -> true end)
+# results = PrimeSieve.run_sieve(primes, 3, size)
+#   |> Enum.drop_every(2)
+#   |> PrimeSieve.count_primes()
+# IO.inspect results
+# IO.inspect PrimeSieve.validate_results(size, results, reference_results)
```
If this commented-out code is not needed for the final solution, then please remove it.

cdesch replied:
Thanks. Removed!

cdesch commented:
Thanks @rbergen ! I pushed the changes you noted!

rbergen commented:
I'm wondering if there is a way (like a command-line option to elixir) to suppress the warning messages about unused variables that are shown when running the implementation?

cdesch commented:
yes, @rbergen or I could just fix the warnings 👍

rbergen commented:
You know what, I think you're right :)

rbergen commented on PrimeElixir/solution_1/prime.ex:
```diff
@@ -0,0 +1,105 @@
+end
+
+# Setup
+size = 1000
```
The sieve size does need to be set to 1,000,000, if possible.

cdesch commented:
@rbergen Thanks. I fixed the `sieve_count`.

I also wrapped everything into a `mix` project that elixir uses typically with tests and updated the docs. 

Running the docker container is exactly as it was before for integration with `make`

rbergen commented:
Do the latest commits mark the end of your work on this?

cdesch commented:
Yes. With the last push I did some final clean up the code (DRY and formatting) and readme.md

It is good to go. Thank you!

I'm sure someone will have some suggestions later on for optimization or fixing a mistake. Elixir doesn't do loops and iterations exactly the same as in Dave's base algorithm, but there might be a happy medium to be as faithful as possible with the language as well as other solutions that are closer to the `wheel` algorithm.  

thank you again for the code review @rbergen

cdesch commented:
One Moment. Ran the docker container again and got an error. I'll fix.

cdesch commented:
All set! Ready to go! Thanks
